components/contrib/kubeflow/pytorch-launcher/sample.py

```diff
@@ -66,124 +103,96 @@ def create_worker_spec(
             },
         }
 
-    worker_spec_output = namedtuple(
-        "MyWorkerOutput", ["worker_spec"]
+    return worker
+
+# container component description setting inputs and implementation
+@dsl.container_component
+def pytorch_job_launcher(
+    name: str,
+    kind: str = "PyTorchJob",
+    namespace: str = "kubeflow",
+    version: str = 'v2',
+    master_spec: dict = {},
+    worker_spec: dict = {},
+    job_timeout_minutes: int = 1440,
+    delete_after_done: bool = True,
+    clean_pod_policy: str = 'Running',
+    active_deadline_seconds: Optional[int] = None,
+    backoff_limit: Optional[int] = None,
+    ttl_seconds_after_finished: Optional[int] = None,
+):
+    command_args = [
+        '--name', name,
+        '--kind', kind,
+        '--namespace', namespace,
+        '--version', version,
+        '--masterSpec', master_spec,
+        '--workerSpec', worker_spec,
+        '--jobTimeoutMinutes', job_timeout_minutes,
+        '--deleteAfterDone', delete_after_done,
+        '--cleanPodPolicy', clean_pod_policy,]
+    if active_deadline_seconds is not None and isinstance(active_deadline_seconds, int):
+        command_args.append(['--activeDeadlineSeconds', str(active_deadline_seconds)])
+    if backoff_limit is not None and isinstance(backoff_limit, int):
+        command_args.append(['--backoffLimit', str(backoff_limit)])
+    if ttl_seconds_after_finished is not None and isinstance(ttl_seconds_after_finished, int):
+        command_args.append(['--ttlSecondsAfterFinished', str(ttl_seconds_after_finished)])
+    
+    return dsl.ContainerSpec(
+        image='quay.io/rh_ee_fwaters/kubeflow-pytorchjob-launcher:v2',
```

Comment:
Hrm, not sure if we should be referencing users individual quay accounts here, but I don't know how we have typically handled such cases in the past. 

Options are: 

1. We add a workflow that builds this image and uploads it to ghcr for the kubeflow repo 
2. We just add a placeholder here, and leave a comment telling the user they need to build/upload this themselves 

since we have the build_image.sh script here, I'm leaning towards (2) and leaving a comment to run this script and replace with the repo here

Reply:
Another option could be to remove the image completely. We could define a lightweight component with [packages_to_install](https://www.kubeflow.org/docs/components/pipelines/user-guides/components/lightweight-python-components/#packages_to_install) and add the `launch_pytorchjob.py` code in there. This way we don't have to build and maintain the image. Happy to be corrected if I'm missing something.

Reply:
So from my understanding this is a sample code. If it's a sample code created by a contributor, I don't see a problem.

My main concern here is if we begin maintaining images from samples it will be a pain for KFP having so many images.

Reply:
I am happy to update it, and add a comment with instructions on using the build script. Is this sufficient or do we want to define a lightweight component which includes the contents of the launch_pytorchjob.py script instead?

Reply:
Updated as per point 2. above. https://github.com/kubeflow/pipelines/pull/11273/commits/1c9ae31d90af072b085a3d5d309bcd437eea0763

Reply:
Here's what I would recommend doing in sample.py:

1. Remove the command arg from pytorch_job_launcher component and directly add the [main](https://github.com/kubeflow/pipelines/blob/1c9ae31d90af072b085a3d5d309bcd437eea0763/components/kubeflow/pytorch-launcher/src/launch_pytorchjob.py#L81) function code from `launch_pytorchjob.py`. This would remove the argument parsers in `launch_pytorchjob.py` thus simplifying it further. It would also eliminate the need for the user to build an image which is not required for just installing `kubeflow-training`. 
2. Remove the `create_master_spec` and `create_worker_spec` as they don't need be their own component. Use util functions like [1](https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L494) and [2](https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/training/api/training_client.py#L469) to create these specs directly in the `pytorch_job_launcher` container component.

Reply:
Please see latest commit https://github.com/kubeflow/pipelines/pull/11273/commits/dbea49b17b1168487a38929ea2ee9c1e019a8453 with refactor as suggested. @Shreyanand @HumairAK @rimolive 
Thanks for the help @Shreyanand :)

Reply:
This looks so much cleaner! Thanks for making the changes @Fiona-Waters 💯 I'm gonna test it out but from the looks of it, this should be good to go.